src/cider/nrepl/middleware/out.clj

```diff
@@ -0,0 +1,86 @@
+(ns cider.nrepl.middleware.out
+  "Change *out* to print on connections in addition to process out.
+
+  Automatically changes the root binding of *out* to print to any
+  active connections. An active connection is a session that has sent
+  at least one \"eval\" op.
+
+  We use an eval message, instead of the clone op, because there's no
+  guarantee that the channel that sent the clone message will properly
+  handle output replies."
+  (:require [cider.nrepl.middleware.util.cljs :as cljs]
+            [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [clojure.tools.nrepl.middleware.interruptible-eval :as ie]
+            [clojure.tools.nrepl.middleware.session :as session])
+  (:import [java.io PrintWriter Writer]))
+
+;;; OutStream
+(defonce original-out *out*)
+
+(declare eval-messages)
+
+(defmacro do-out-messages
+  "Run body with v bound to the output stream of each msg in msg-seq.
+  Also run body with v bound to `original-out`."
+  [[v msg-seq] & body]
+  `(do (let [^Writer ~v original-out]
+         ~@body)
+       (doseq [{:keys [~'session] :as ~'msg} ~msg-seq]
+         (let [^Writer ~v (get @~'session #'*out*)]
+           (try (binding [ie/*msg* ~'msg]
+                  ~@body)
+                ;; If a channel is faulty, dissoc it.
+                (catch Exception ~'e
+                  (swap! eval-messages dissoc (:id (meta ~'session)))))))))
+
+(defn fork-out
+  "Returns a PrintWriter suitable for binding as *out* or *err*. All
+  operations are forwarded to all output bindings in the sessions of
+  messages in addition to the server's usual PrintWriter (saved in
+  `original-out`)."
+  [messages]
+  (PrintWriter. (proxy [Writer] []
+                  (close [] (.flush ^Writer this))
+                  (write [& [x ^Integer off ^Integer len]]
+                    (do-out-messages [out messages] (.write out x off len)))
+                  (flush []
+                    (do-out-messages [out messages] (.flush out))))
+                true))
+
+;;; Known eval sessions
+(def eval-messages
```

Comment:
I find this name a bit confusing, but I can't propose a great alternative. Perhaps something along the lines of `tracked-sessions-map`?